### DISCUSSION TITLE
Using Luge with Astro + Svelte

### DISCUSSION THREAD

**Discussion by TomSmedley** [AntoineW/luge#19] [Q&A]
Hey,

I'm trying to use Luge with Astro and Svelte - I was wondering what was the best approach? As Luge automatically initiate any elements with the write data attributes, the import logic doesn't seem to make sense when it comes to Astro which is very component based. With Svelte having `use` with it's cycle of initiation and destroying, is there any plan to bring in manual initialisation using a Node element? For example `Luge.parallax(node)`. 

Thanks,
Tom

**Comment by AntoineW** [comment]
Hey!
I don't know Astro and Svelte but I assume it has a similar logic as other JS frameworks? So, emitting an `update` event like so `luge.emitter.emit('update')` after components rendering could do the trick.

Anyway, yes adding a proper API to be able to use luge without using the data attributes is in my to-do list!
